src/Rules/File/FileExtensionRule.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TwigCsFixer\Rules\File;
+
+use TwigCsFixer\File\FileHelper;
+use TwigCsFixer\Rules\AbstractRule;
+use TwigCsFixer\Rules\ConfigurableRuleInterface;
+
+/**
+ * Ensures file name uses two extensions (e.g. index.html.twig).
+ */
+final class FileExtensionRule extends AbstractRule implements ConfigurableRuleInterface
+{
+    /**
+     * @param array<string> $ignoredSubDirectories
+     */
+    public function __construct(
+        private ?string $baseDirectory = null,
+        private array $ignoredSubDirectories = [],
+    ) {
+    }
+
+    public function getConfiguration(): array
+    {
+        return [
+            'baseDirectory'         => $this->baseDirectory,
+            'ignoredSubDirectories' => $this->ignoredSubDirectories,
+        ];
+    }
+
+    protected function process(int $tokenPosition, array $tokens): void
+    {
+        if (0 !== $tokenPosition) {
+            return;
+        }
+
+        $token = $tokens[$tokenPosition];
+        $fileName = FileHelper::getFileName(
+            $token->getFilename(),
+            $this->baseDirectory,
+            $this->ignoredSubDirectories,
+        );
+        if (null === $fileName) {
+            return;
+        }
+
+        $fileParts = explode('.', $fileName);
+        $fileExtension = array_pop($fileParts);
+        if ('twig' !== $fileExtension) {
+            // Not a Twig file: skip
+            return;
+        }
+
+        $formatExtension = array_pop($fileParts);
+        if ([] === $fileParts || null === $formatExtension) {
```

Comment:
I'm not sure about the `[] === $fileParts`.

If understand correctly the checks, it means the file is `.foo.twig`, then `foo` is not "the second extension" but the name of the dotfile. Should we report an error for this ? What if someone use `.gitignore.twig` like https://github.com/sonata-project/dev-kit/blob/master/templates/project/.gitignore.twig but maybe they just should ignore the rule because they also have `Makefile.twig` https://github.com/sonata-project/dev-kit/blob/master/templates/project/Makefile.twig...

Let's keep this check then.
But can you add a test `.dotfile.twig` and `.dotfile.html.twig` ?